CoreAlgorithmPlugins/src/au/gov/asd/tac/constellation/plugins/algorithms/sna/centrality/PagerankCentralityPlugin.java

```diff
@@ -95,118 +92,181 @@ public PluginParameters createParameters() {
-                            for (int transactionPosition = 0; transactionPosition < transactionCount; transactionPosition++) {
-                                final int transactionId = graph.getVertexTransaction(neighbourId, transactionPosition);
-                                final int transactionDirection = graph.getTransactionDirection(transactionId);
-                                final int transactionSourceId = graph.getTransactionSourceVertex(transactionId);
-                                final int transactionDestinationId = graph.getTransactionDestinationVertex(transactionId);
-                                if (transactionSourceId != transactionDestinationId) {
-                                    final int otherId = neighbourId == transactionSourceId ? transactionDestinationId : transactionSourceId;
-                                    if ((treatUndirectedBidirectional && transactionDirection == GraphConstants.FLAT)
-                                            || (neighbourId < otherId && transactionDirection == GraphConstants.UPHILL)
-                                            || (neighbourId > otherId && transactionDirection == GraphConstants.DOWNHILL)) {
-                                        neighbourOutCount++;
-                                    }
-                                }
-                            }
+        }
+
+        // update the graph with pagerank values
+        PagerankVertex.commitPageranksToGraph();
+    }
+    
+    static class PagerankVertex {
+        private static GraphWriteMethods graph;
+        private static final Set<PagerankVertex> sinks = new HashSet<>();
+        private static boolean treatUndirectedBidirectional;
+        private static boolean normaliseByAvailable;
+        private static double dampingFactor = 0;
+        private static int vertexCount = 0;
+        private static PagerankVertex[] pagerankVertices;
+        private static double sinkPagerankContribution = 0;
+        private static double delta = 0;
+        private static int pagerankAttribute;
+        private static double baseContribution;
+        private final int vertexId;
+        private final Set<PagerankVertex> neighbours = new HashSet<>();
+        private int outCount = 0;
+        private double pagerank;
+        private double pagerankContribution;
+        private double stagedPagerank;
+        private boolean isSink = false;
 
-                            vertexInLinks.get(vertexId).add(Tuple.create(neighbourId, neighbourOutCount));
+        private PagerankVertex(int vertexPosition) {
+            this.vertexId = graph.getVertex(vertexPosition);
+            this.pagerank = 1.0/vertexCount;
+            this.pagerankContribution = 0;
+        }
+        
+        static void initialiseAllPagerankVertices(final GraphWriteMethods graph, final boolean treatUndirectedBidirectional, final double dampingFactor, final boolean normaliseByAvailable) {
```

Comment:
Could this be made private or was the intention that this could be called elsewhere?

Reply:
I can (and will) but what is the advantage over it being package private, given its in a private class? Is it just a legibility thing?

Reply:
I think it's definitely a legibility thing for me. Even though it's perfectly valid, it just looks odd not having a (visible) access modifier. Given it is in a private class, so long as the function isn't private it probably doesn't make much of difference what access modifier you add to it (package private, protected, public) in this instance. Really I'm just suggesting what I've seen elsewhere